Index: mold/Makefile
===================================================================
--- mold.orig/Makefile
+++ mold/Makefile
@@ -135,7 +135,7 @@ out/git-hash.o: out/git-hash.cc
 	$(CXX) $(MOLD_CXXFLAGS) $(CXXFLAGS) -c -o $@ $<
 
 mold: $(OBJS) $(MIMALLOC_LIB) $(TBB_LIB)
-	$(CXX) $(OBJS) -o $@ $(MOLD_LDFLAGS) $(LDFLAGS)
+	$(CXX) $(OBJS) -pie -o $@ $(MOLD_LDFLAGS) $(LDFLAGS)
 	ln -sf mold ld
 	ln -sf mold ld64
 
