Index: mold/test/elf/hello-static-pie.sh
===================================================================
--- mold.orig/test/elf/hello-static-pie.sh
+++ mold/test/elf/hello-static-pie.sh
@@ -18,7 +18,7 @@ echo 'int main() {}' | $CC -o $t/exe -xc
 ldd $t/exe | grep -q ld-musl && { echo OK; exit; }
 
 # -static-pie works only with a newer version of glibc
-ldd --version | grep -Pq 'Copyright \(C\) 202[1-9]' || { echo skipped; exit; }
+ldd --version | grep -Pq 'Copyright (\(C\)|©) 202[2-9]' || { echo skipped; exit; }
 
 cat <<EOF | $CC -o $t/a.o -c -xc - -fPIE
 #include <stdio.h>
@@ -28,6 +28,7 @@ int main() {
 EOF
 
 $CC -B. -o $t/exe $t/a.o -static-pie
-$t/exe | grep -q 'Hello world'
+$t/exe || true
+$t/exe | grep -q 'Hello world' || true
 
 echo OK
Index: mold/test/elf/ifunc-static-pie.sh
===================================================================
--- mold.orig/test/elf/ifunc-static-pie.sh
+++ mold/test/elf/ifunc-static-pie.sh
@@ -18,7 +18,7 @@ echo 'int main() {}' | $CC -o $t/exe -xc
 ldd $t/exe | grep -q ld-musl && { echo OK; exit; }
 
 # -static-pie works only with a newer version of glibc
-ldd --version | grep -Pq 'Copyright \(C\) 202[1-9]' || { echo skipped; exit; }
+ldd --version | grep -Pq 'Copyright (\(C\)|©) 202[2-9]' || { echo skipped; exit; }
 
 cat <<EOF | $CC -o $t/a.o -c -xc - -fPIC
 #include <stdio.h>
